### ISSUE TITLE
Question on gradient scaling step in code

### ISSUE THREAD

**Issue by sp9103** [prs-eth/Marigold-DC#8]
Hello, 
Thank you for releasing the code and paper Marigold-DC.

I have a question about the following code block:
``` python
# Scale gradients up
with torch.no_grad():
    pred_epsilon_norm = torch.linalg.norm(pred_epsilon).item()
    depth_latent_grad_norm = torch.linalg.norm(pred_latent.grad).item()
    scaling_factor = pred_epsilon_norm / max(depth_latent_grad_norm, 1e-8)
    pred_latent.grad *= scaling_factor
```

Which part of the paper does this gradient scaling correspond to?
Is this scaling an additional implementation detail for stability, or is it implicitly covered in the paper’s formulation?

Thank you for your clarification!
